### PR TITLE
Remove React-Native and CocoaPods

### PR DESCRIPTION
**Pull request by szabinst**
refs: MBL-17681
affects: Student, Parent, Teacher
release note: none
test plan: Test everything and beyond

## What's changed
- Removed React-Native
- Moved remaining Firebase dependencies from CocoaPods to SPM
- Deintegrated CocoPods from the project
- As part of a follow up initiative I started reorganizing the project root folder by moving documents into `./github` folder. 

## Benchmark data
- Switching branches is now 13 times faster on average. Now it's consistently around 1,43s while before it sometimes took 33s. 
- Clean build is now 39% faster on average.
- 3rd party dependency management tool count is reduced to 2 from 3.
- Almost 10 times less 3rd party dependencies. 
- Codebase is reduced by 27,9% which equals to 254 072 less lines of code. 

## What to test
- Compile build and test targets
- Test all the apps

## What's next
- Since we have a bit of an unusual project setup by depending on yarn for automations, this creates a `node_modules` folder in the project root. I'll create a follow-up PR where I'll move iOS related source files into a dedicated folder called `source`. 
- We'll be able to create xcodegen yml files for Parent and Teacher.

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created
- [ ] A11y checked
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode
- [ ] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a release note. If no release note is wanted, use `none`. Example: `release note: Fixed a bug that prevented users from enjoying the app.`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add which apps this change affects. Example: `affects: Teacher, Student` or `affects: none`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a reference to a Jira ticket. For example: `refs: MBL-10023`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 10797bb5ef43f6085b2ae977eda23232b835543c
</p>

**Comment by szabinst** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by szabinst** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by szabinst** [comment]
<details><summary>Parent Build QR Code:</summary>

</details>

File: Student/Student/StudentAppDelegate.swift [hunk 1]
```diff
@@ -143,22 +141,37 @@ class StudentAppDelegate: UIResponder, UIApplicationDelegate, AppEnvironmentDele
...
-                        NativeLoginManager.login(as: session)
                     }
                     .store(in: &self.subscriptions)
             }
+
+            self.setTabBarController()
         }}
     }
 
+    func setTabBarController() {
```
Comment (rh12) [inline Student/Student/StudentAppDelegate.swift:152]: Not necessarily in this PR, but would it make sense to implement a superclass like `BaseAppDelegate` to handle the common parts?

**Review by rh12 — COMMENTED** [review]
It's really great to see this go :)

Code review + 1

File: Student/Student/StudentAppDelegate.swift (hunk 1 above)
Comment (szabinst) [reply Student/Student/StudentAppDelegate.swift:152]: Yes, it's on the list of future improvements but out of scope of this PR.

**Review by vargaat — COMMENTED** [review]
The build-icons script still creates a react file, otherwise looks good!

**Comment by rh12** [comment]
This is a minor thing, and maybe unrelated, but the default blue color changed to a lighter in the Calendar:
<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

It turned back to the old blue after I changed user and changed back.

**Comment by szabinst** [comment]
@vargaat fixed. 
@rh12 I think this is unrelated.

**Review by vargaat — APPROVED** [review]

**Review by rh12 — APPROVED** [review]
QA + 1

**Pull request merged**